controller/backing_image_controller.go

```diff
@@ -310,22 +345,530 @@ func (bic *BackingImageController) syncBackingImage(key string) (err error) {
+			v2CopiesFailed = append(v2CopiesFailed, v2DiskUUID)
+			continue
+		}
+		if copyStatus.State == longhorn.BackingImageStateUnknown {
+			v2CopiesUnknown = append(v2CopiesUnknown, v2DiskUUID)
+			continue
+		}
+	}
+
+	for _, v2DiskUUID := range missingSpecV2Copies {
+		node, _, err := bic.ds.GetReadyDiskNodeRO(v2DiskUUID)
+		if err != nil {
+			return errors.Wrapf(err, "failed to get the ready disk node for disk %v", v2DiskUUID)
+		}
+
+		engineClientProxy, err := bic.getEngineClientProxy(node.Name, longhorn.DataEngineTypeV2, log)
+		if err != nil {
+			return errors.Wrapf(err, "failed to get the engine client proxy for node %v", node.Name)
+		}
+
+		log.Infof("Deleting the missing spec v2 copy on disk %v", v2DiskUUID)
+		err = engineClientProxy.SPDKBackingImageDelete(bi.Name, v2DiskUUID)
+		engineClientProxy.Close()
+		if err != nil {
+			return errors.Wrapf(err, "failed to delete the v2 backing image on diskUUID %v", v2DiskUUID)
+		}
+
+		delete(bi.Status.DiskFileStatusMap, v2DiskUUID)
+		bic.v2CopyBackoff.DeleteEntry(v2DiskUUID)
+		bic.eventRecorder.Eventf(bi, corev1.EventTypeNormal, constant.EventReasonDelete, "Deleted failed v2 backing image %v on disk %v on node %v", bi.Name, v2DiskUUID, node.Name)
+	}
+
+	// When the backing image is created on that disk, it is either ready, failed or unknown.
+	// We directly delete the failed v2 copy if there is ready copy
+	// For unknown v2 copy, it can happen in the following case,
+	// 1. If the instance-manager pod is not running or the instance-manager CR is not ready.
+	// 2. Backing image copy has status (ready or failed) before but can't get the ststus now.
+	// 3. Backing image copy is failed to be created on the disk and there is no lvol snapshot and at the same time the node is rebooted.
+	//    It will keeps in unknown state forever because spdk server has no snapshot lvol to pick up to reconstruct the record.
+	// For case 1 and case 2, the status will come back automatically when server pick the snapshot lvol up and reconstruct the record.
+	// We delete the unknown copy only for the case 3 if there is a ready copy so it can be recreated.
+	if !hasReadyV2BackingImage {
+		return nil
+	}
+
+	for _, v2DiskUUID := range v2CopiesUnknown {
+		node, _, err := bic.ds.GetReadyDiskNodeRO(v2DiskUUID)
+		if err != nil {
+			return errors.Wrapf(err, "failed to get the ready disk node for disk %v", v2DiskUUID)
+		}
+
+		isDeleted, err := bic.deleteUnknownV2Copy(bi, v2DiskUUID, node, log)
+		if err != nil {
+			return errors.Wrapf(err, "failed to delete the unknown v2 copy on disk %v", v2DiskUUID)
+		}
+		// lvol is found so we didn't delete the unknown copy
+		if !isDeleted {
+			continue
+		}
+
+		delete(bi.Status.DiskFileStatusMap, v2DiskUUID)
+		bic.v2CopyBackoff.DeleteEntry(v2DiskUUID)
+		bic.eventRecorder.Eventf(bi, corev1.EventTypeNormal, constant.EventReasonDelete, "Deleted unknown v2 backing image %v on disk %v on node %v", bi.Name, v2DiskUUID, node.Name)
+	}
```

Comment:
nit: abstract it to `cleanupUnknowSpecV2Copies()`

Reply:
fixed.